plugins/check-path/core-rs/src/paths.rs

```diff
@@ -0,0 +1,84 @@
+use js_sys::Reflect;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+use wasm_bindgen::prelude::*;
+
+enum NodeValue {
+    StringType(String),
+    ObjectType(JsValue),
+    None,
+}
+
+pub struct Paths {
+    store: Rc<RefCell<HashMap<String, Vec<String>>>>,
+}
+
+impl Paths {
+    pub fn new() -> Self {
+        Self {
+            store: Rc::new(RefCell::new(HashMap::new())),
+        }
+    }
+
+    pub fn parse(&self, data: JsValue) {
+        self.traverse(data);
+    }
+
+    pub fn get(&self, key: &str) -> Vec<String> {
+        self.store
+            .borrow()
+            .get(key)
+            .or(Some(&vec!["".to_string()]))
+            .unwrap()
+            .clone()
```

Comment:
Also look into [Ref::map](https://doc.rust-lang.org/std/cell/struct.Ref.html#method.map) to see if you can return a borrowed value instead of cloning

Reply:
If I understand the docs, at a quick glance (so probably not really getting it), I would have to wrap up the vector in an RefCell in order to get a `Ref<Vec<String>>` from ` self.key_paths.borrow().get(key).borrow()` ??